src/loaders/OmeZarrLoader.ts

```diff
@@ -178,139 +216,201 @@ function pickLevelToLoad(loadSpec: LoadSpec, multiscaleDims: number[][], dimInde
-    this.cache = cache;
+  private constructor(
+    scaleLevels: ZarrArray[],
+    multiscaleMetadata: OMEMultiscale,
+    omeroMetadata: OmeroTransitionalMetadata,
+    axisTCZYX: [number, number, number, number, number]
+  ) {
+    this.scaleLevels = scaleLevels;
+    this.multiscaleMetadata = multiscaleMetadata;
+    this.omeroMetadata = omeroMetadata;
+    this.axesTCZYX = axisTCZYX;
   }
 
-  async loadDims(loadSpec: LoadSpec): Promise<VolumeDims[]> {
-    const store = new HTTPStore(this.url);
-
-    const data = await openGroup(store, null, "r");
+  static async createLoader(url: string, scene = 0, cache?: VolumeCache): Promise<OMEZarrLoader> {
+    // Setup: create store, get basic metadata
+    const store = new ChunkCachingStore(new HTTPStore(url));
+    const group = await openGroup(store, null, "r");
+    const metadata = (await group.attrs.asObject()) as OMEZarrMetadata;
 
-    // get top-level metadata for this zarr image
-    const allmetadata = await data.attrs.asObject();
-    // each entry of multiscales is a multiscale image.
-    const imageIndex = imageIndexFromLoadSpec(loadSpec, allmetadata.multiscales);
-    const datasets: OMEDataset[] = allmetadata.multiscales[imageIndex].datasets;
-    const axes: Axis[] = allmetadata.multiscales[imageIndex].axes;
+    // Pick scene (multiscale)
+    if (scene > metadata.multiscales.length) {
+      console.warn(`WARNING: OMEZarrLoader: scene ${scene} is invalid. Using scene 0.`);
+      scene = 0;
+    }
+    const multiscale = metadata.multiscales[scene];
+
+    // Open all scale levels of multiscale
+    const scaleLevelPaths = multiscale.datasets.map(({ path }) => path);
+    const scaleLevelPromises = scaleLevelPaths.map((path) => openArray({ store, path, mode: "r" }));
+    const scaleLevels = await Promise.all(scaleLevelPromises);
+    const dimensionSeparator = scaleLevels[0].meta.dimension_separator;
+
+    // Grab axis indexes, use them to set up cache store
+    const axisTCZYX = remapAxesToTCZYX(multiscale.axes);
+    const [t, c, z, y, x] = axisTCZYX;
+    const numChannels = Math.max(scaleLevels[0].shape[c], 1);
+    const numTimes = Math.max(scaleLevels[0].shape[t], 1);
+    const scaleChunkDims = scaleLevels.map(
+      ({ chunkDataShape }) => new Vector3(chunkDataShape[x], chunkDataShape[y], chunkDataShape[z])
+    );
+    const cacheStore = cache?.addVolume(numChannels, numTimes, scaleChunkDims);
+    store.setCacheInfo(axisTCZYX, scaleLevelPaths, dimensionSeparator, cache, cacheStore);
 
-    const axisTCZYX = remapAxesToTCZYX(axes);
+    return new OMEZarrLoader(scaleLevels, multiscale, metadata.omero, axisTCZYX);
+  }
 
-    // Assume all axes have the same units - we have no means of storing per-axis unit symbols
-    const spaceUnitName = axes[axisTCZYX[4]].unit;
+  private getUnitSymbols(): [string, string] {
+    // Assume all spatial axes have the same units - we have no means of storing per-axis unit symbols
+    const xi = this.axesTCZYX[4];
```

Comment:
Okay this is maybe a silly suggestion by my brain keeps thinking it would be really nice to have an enum for TCZYX in case this ever changes